### PR TITLE
[NLU-3981] - Failing Catalan "quarter-to" regression tests

### PR DESCRIPTION
**Pull request by BobbyMartins5**
Fixed catalan "quarter to" regression bugs, modified tests

### CODE REVIEW

**Comment by BobbyMartins5** [comment]
**Note**: with the current implementation, "Ara és un quart de nou" (It's now a quarter to nine) resolves as both the current time and 08:45:00 + 20:45:00. This is how it is resolved for english as well.

File: Specs/DateTime/Catalan/TimeParser.json [hunk 1]
```diff
@@ -120,42 +120,42 @@
...
       }
     ]
   },
   {
-    "Input": "un quart de nou",
+    "Input": "Hi ha vol cap a un quart de 14:00?",
     "Results": [
       {
-        "Text": "quart de nou",
+        "Text": " un quart de 14",
```
Comment (colm-dillon) [inline Specs/DateTime/Catalan/TimeParser.json:186]: @KanchanKumar12 and @richcamp  correct me if I'm wrong but wasn't the native speakers saying this was actually 13:15 - "one quarter of 2" .They brought up the example of 3 quarters of 2 which would be 13:45? Presume it's the same in this use case.

**Review by colm-dillon — APPROVED** [review]

**Comment by BobbyMartins5** [comment]
At the beginning of this task, there was an issue with the phrase "un quart de" not working. Initially this phrase was considered to mean "a quarter to". 

`un quart de nou` = `08:45:00` \ `20:45:00`
`un quart de 14:00` = `13:45:00`

The issue was caused by a missing regex entry for 'un' in the `LessThanOneHour` regex and 'de' in the `TimePrefix` regex. There was also missing logic for dealing with 'un quart de' in `time_parser_config.py`. These were all resolved in [this pr](https://github.com/purecloudlabs/Recognizers-Text/pull/108/files)

This fix address comments made that 'un quart de' should resolve as 'a quarter past' instead. 

`un quart de nou` = `08:15:00` \ `20:15:00`

It also adds support for the following terms:
`Són tres quarts de sis` = `It is quarter to six` (05:45:00)
`Són dos quarts de vuit` = `It is half past eight` (07:30:00)

See [here](https://www.omniglot.com/language/time/catalan.htm) and [here](https://en.wikipedia.org/wiki/Catalan_time_system) for context about this system

Also modified some of the tests to reflect these changes.

**Comment by BobbyMartins5** [comment]
Successful Pipeline run ✅: https://jenkins.ininica.com/job/NLU/job/Recognizers-Text/347/

**Review by andrew-gradinari — APPROVED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/time_parser_config.py [hunk 2]
```diff
@@ -53,15 +53,18 @@ def adjust_by_prefix(self, prefix: str, adjust: AdjustParams):
...
-            delta_min = -15
-        elif prefix.startswith('quart') or prefix.startswith('i quart'):
+        if prefix.startswith('menys quart') or prefix.startswith('quarts') or prefix.startswith('tres quarts') or \
+                prefix.startswith('quart menys'):
+            delta_min = -45
+        elif prefix.startswith('i quart'):
             delta_min = 15
-        elif prefix.startswith('mitjana') or prefix.startswith('i mitjana') or \
-                prefix.startswith('i mitja') or prefix.startswith(('mitja')):
+        elif prefix.startswith('quart') or prefix.startswith('un quart'):
```
Comment (samhickey25) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/time_parser_config.py]: are these right?

**Review by kevinwalshgen — APPROVED** [review]

**Review by Conor-Keaney — APPROVED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/time_parser_config.py (hunk 2 above)
Comment (BobbyMartins5) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/time_parser_config.py]: tres menys quart - 03:45
dos quart menys - 02:15 (this one is in fact incorrect here)
tres quarts d'una - 12:45
2 i quart - 02:15
(un) quart de nou - 09:15
Rest are correct as far as I've seen

**EDIT**: for quart menys, it is most commonly used in the following format: tres quarts menys cinc de quatre (three quarters minus five of four) = 03:40:00

I think for now, I will leave quart menys to resolve as xx:45:00, so as in a future ticket this functionality can be implemented if needs be.

**Pull request merged**